### ISSUE TITLE
Standardised messages and docs

### ISSUE THREAD

**Issue by jeddy3** [stylelint/stylelint#46]
What do you guys think about standardising the wording within the messages? They've diverged a little bit already. 

The "Expected" approach that @davidtheclark's [used in](https://github.com/stylelint/stylelint/blob/master/src/rules/declaration-bang-space/index.js#L9) `declaration-bang-space` and [in](https://github.com/stylelint/stylelint/blob/master/src/rules/declaration-block-trailing-semicolon/index.js#L4) `declaration-block-trailing-semicolon` LGTM. It seems to be used a lot in eslint as well.

So, the current rules would look something like:

Note: always tending towards "expected" over "unexpected"...

* `declaration-colon/bang-space`: "Expected single space before "!"" and "Expected no space before "!"
* `declaration-block-trailing-semicolon`: "Expected a trailing semicolon" and "Expected no trailing semicolon"
* `number-leading-zero`: "Expected a leading zero for fractional value less than 1" and "Expected no leading zero for fractional value less than 1" 

And "unexpected" would be mainly used for `*-no-*` rules e.g.

* `declaration-no-important`: "Unexpected !important"
* `rule-set-no-single-line`: "Unexpected single-line rule-set"

What do you think?

I'm still keen to keep the *must always*, *must never*, *can*, *disallow*, and *enforce* language in the   documentation as I think it feels nice and clear. e.g. [here](https://github.com/stylelint/stylelint/blob/declaration-colon-before/src/rules/declaration-colon-space/README.md) and [here](https://github.com/stylelint/stylelint/blob/declaration-no-important-options/src/rules/declaration-no-important/README.md).

Thoughts?

**Comment by davidtheclark** [comment]
Sounds good. One question: For e.g. `number-leading-zero` should the negative instead use "unexpected": "Unexpected leading zero for fractional value less than 1"??

**Comment by jeddy3** [comment]
> One question: For e.g. number-leading-zero should the negative instead use "unexpected": "Unexpected leading zero for fractional value less than 1"??

Yeap, I had that same thought just after I clicked "Comment"!  :) I

That works for me if we make the other "always"|"never" rules consistent too e.g.

* `declaration-colon/bang-space`: "Expected single space before "!"" and "Unexpected space before "!"
* `declaration-block-trailing-semicolon`: "Expected a trailing semicolon" and "Unexpected trailing semicolon"
* `number-leading-zero`: "Expected a leading zero for fractional value less than 1" and "Unexpected leading zero for fractional value less than 1" 

When it comes to rules with values other than "always" and "never" I think we should tend towards "expected" over "unexpected" e.g.

```js
url-quotes: "single"|"double"|none
```

We might need to create multiple rejected messages for each value if we use "unexpected"  e.g.

For `url-quotes: "single"`

We'd need two rejected messages:

* "Unexpected double quotes for URL"
* "Unexpected lack of quotes for URL"

Where as, the "expected" approach would require just one rejected message for each value:

e.g. for `url-quotes: "single"`

We'd only need:

* "Expected single quotes for URL"

Sound good?

**Comment by davidtheclark** [comment]
Yes. I think "unexpected" (aka "rejected") is for situations where what's there is not ok but there is not a specific expected alternative. So `url-quotes` never "rejects" if it's on, because it's always expecting something specific. We're in agreement.

**Comment by jeddy3** [comment]
> We're in agreement.

Good stuff :)

**Comment by MoOx** [comment]
:+1:

**Comment by jeddy3** [comment]
I can do this.
